### DISCUSSION TITLE
convert: width or height exceeds limit error

### DISCUSSION THREAD

**Discussion by artincube** [photoprism/photoprism#3528] [Q&A]
I have some issues indexing some things. One at a time.
I may have wrong settings.
Here one: 
I have
PHOTOPRISM_ORIGINALS_LIMIT: -1
PHOTOPRISM_RESOLUTION_LIMIT: 900

Other similar (or bigger) format size and camera images have indexed properly but
not at least one jpg 34.2 MB panorama shot with a Samsung S7 Model: SM-G930U
16384x3184 pixels.

I don't seem to exceed limits.
But log lines below

2023-07-07 10:20:25 ERRO index: failed adding main jpg file 'S7s 2017-2020/2019/09/20190901_071256.jpg'
2023-07-07 10:20:25 ERRO index: failed creating thumbnails for 'S7s 2017-2020/2019/09/20190901_071256.jpg' (convert: width or height exceeds limit `/photoprism/originals/S7s 2017-2020/2019/09/20190901_071256.jpg' @ error/cache.c/OpenPixelCache/3909. convert: no images defined `/photoprism/storage/cache/media/3/b/8/3b8e64b81eb7a1c92367e5e8a10f035b4a013b45.jpg' @ error/convert.c/ConvertImageCommand/3229. )
2023-07-07 10:20:25 INFO convert: re-encoding 'S7s 2017-2020/2019/09/20190901_071256.jpg' to 3b8e64b81eb7a1c92367e5e8a10f035b4a013b45.jpg (convert)

**Comment by lastzero** [comment]
Can you provide us with a sample file for testing?

https://www.photoprism.app/contact#file-samples

Thank you! ❤️

**Reply by artincube** [reply]
I checked it with GM and it said "unsupported marker type 0x7f"
I used GM to convert the file to another jpeg (the trick in another thread here) setting up quality high (default is 75) and it kept all other metadata. This one properly indexed.
Shall I still upload the original file?

**Reply by artincube** [reply]
I got a big dozen files with something bad that could be saved by graphicsmagick and one that couldn't (GM specified corrupt jpeg data for this one), out of 7,500 S7 files. Do you want them?
I mean the big dozen.

**Reply by NiklasRosenstein** [reply]
I have the same error message with a file a JPEG original file sized `1280x37640`. Unfortunately I can't provide this image as a sample. I imagine any JPEG of that dimension would do to reproduce.

```
photoprism  | time="2023-07-08T17:45:50Z" level=error msg="index: failed creating thumbnails for ecbb118e505e5aed549ddf8c4d9afcc9.jpeg (convert: width or height exceeds limit `/photoprism/originals/ecbb118e505e5aed549ddf8c4d9afcc9.jpeg' @ error/cache.c/OpenPixelCache/3909.\nconvert: no images defined `/photoprism/storage/cache/media/3/6/8/3686211ea9067aa35c9b6f4ce271654f483b194f.jpg' @ error/convert.c/ConvertImageCommand/3229.\n)
```

Is there a hard limit for the file size that Photoprism accepts to converting images, and can it be increased?

**Reply by artincube** [reply]
PHOTOPRISM_ORIGINALS_LIMIT: -1               # file size limit for originals in MB (increase for high-res video) -1 no limit
PHOTOPRISM_RESOLUTION_LIMIT: -1.         # resolution limit in MP -1 no limit

Hi Niklas
Is your image shot with a Samsung, Panorama? All I had this issue with are.
I ended up using GraphicsMagick to make a new jpeg out of them, without issue.
I also tried to open one, that would not be processed by GM, in Photoshop or Gimp, whatever, resave another jpeg with quality above 90 to loose a minimum, and use exiftool to inject metadata from original image.
exiftool -TagsFromFile srcimage.jpg "-all:all>all:all" targetimage.jpg

**Reply by lastzero** [reply]
There's also a limit of 30k pixels in one dimension, so a width of 37k pixels won't work. Note the JPEG decoder is provided by Google as part of the Go programming language. It's not something we have developed, meaning we can't change it.

**Reply by artincube** [reply]
So the only solution is to make a tiff out of the oversize jpeg and inject original metadata to the tiff using exiftool.
Then index it

**Reply by lastzero** [reply]
PNG may work too and could offer better compression. More details can be found in the public Go docs.
